samples/telegram_bot_service/telegram_bot_service.cpp

```diff
@@ -0,0 +1,182 @@
+#include <userver/testsuite/testsuite_support.hpp>
+#include <userver/utest/using_namespace_userver.hpp>
+
+#include <userver/clients/dns/component.hpp>
+#include <userver/clients/http/component.hpp>
+#include <userver/components/component.hpp>
+#include <userver/components/minimal_server_component_list.hpp>
+#include <userver/fs/read.hpp>
+#include <userver/server/handlers/tests_control.hpp>
+#include <userver/utils/daemon_run.hpp>
+#include <userver/yaml_config/merge_schemas.hpp>
+
+#include <userver/telegram/bot/client/client.hpp>
+#include <userver/telegram/bot/components/client.hpp>
+#include <userver/telegram/bot/components/long_poller.hpp>
+
+#include <codecvt>
+#include <locale>
+
+namespace samples::telegram_bot {
+
+class GreetUser final : public telegram::bot::TelegramBotLongPoller {
+ public:
+  static constexpr std::string_view kName = "handler-greet-user";
+
+  GreetUser(const components::ComponentConfig& config,
+            const components::ComponentContext& context);
+  
+  void HandleUpdate(telegram::bot::Update update,
+                    telegram::bot::ClientPtr client) override;
+  
+  static yaml_config::Schema GetStaticConfigSchema();
+
+ private:
+  std::int64_t SendGreeting(
+      telegram::bot::ClientPtr client,
+      std::int64_t chat_id,
+      const std::optional<std::string>& username) const;
+
+  static std::string GreetingCaption(
+      const std::optional<std::string>& username);
+
+  static std::vector<telegram::bot::MessageEntity> GreetingMessageEntities(
+      std::string_view caption);
+
+  telegram::bot::SendPhotoMethod::Parameters::Photo GreetingPhoto() const;
+
+  static std::shared_ptr<std::string> ReadPhoto(
+      const components::ComponentConfig& config,
+      const components::ComponentContext& context);
+
+  telegram::bot::RequestOptions request_options_;
```

Comment:
This field could be moved inside of components clients and provided within [client.hpp](https://github.com/userver-framework/userver/pull/385/files#diff-07cd03d55832104394bc8b660ced8c6a8e582298e8381126b1ad11172b92292dR32).
It looks statically configurable and may be provided by ComponentConfig or dynamic_config